features/202-session-management/api.md

```diff
@@ -11,19 +12,29 @@ interface UserAgent {
     device: string;
 }
 
+// Field value would be empty string if not available
+interface Device {
+    name: string; // device name
+}
+
 interface Session {
     id: string;
     identityID: string;
     createdAt: Date;
     lastAccessedAt: Date;
     createdByIP: string;
     lastAccessedIP: string;
-    createdByUserAgent: UserAgent;
-    lastAccessedUserAgent: UserAgent;
+    userAgent: UserAgent;
+    device: Device?;
     name: string;
     data: object;
 }
 
+interface ExtraSessionInfoConfig {
```

Comment:
Actually I think having an `ExtraSessionInfoConfig` is very strange. Why does DeviceName an `ExtraSessionInfo` but not other fields? Why we need such a distinguishment?

Reply:
Moreover, if we somehow collect even more sensitive information for each session for security reason, the opt-in / opt-out would be per information pieces. So it also doesn't make sense to have a "global switch" to enabled or disable `ExtraSessionInfoConfig`

Reply:
I use browser user agent as the baseline; mobile SDKs are expected to send information contained in browser user agent in user agent header. Extra information are put in the cookie and configure here.

Reply:
We can have a granular toggle for individual information pieces, and a global 'kill switch' to disable all extra info collection (i.e. `enabled`).

Reply:
1. Please, I really hate the idea of requiring cookies just for session information even for token-based header. It adds a very big hassle for client SDK implementation just for a piece of information...
2. I think the baseline of browser user agent's information is vague and an unnatural concept to follow for developer users. What are the downside of sticking with a simple Session Data structure?
3. I think having two levels of feature switch here is too complicated. We won't have a lot of switch anyway, keep it simple? (The only one other than device name I can foresee is more accurate location e.g. GPS which obviously need to be opt-in.)

Reply:
1. Okay, then mobile SDKs would not support cookie as session token transport.
2. It is not expected for normal developer to understand the format of user agent. It is generated by us and consumed by us, also for intermediate consumers like analytics and firewalls.
3. Okay, I'll use individual switches for each information pieces.